http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -618,29 +616,29 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
       )
     } yield ()
 
-  def handleCreateAppMessage(msg: CreateAppMessage)(
+  private[monitor] def handleCreateAppMessage(msg: CreateAppMessage)(
     implicit ev: ApplicativeAsk[F, AppContext]
   ): F[Unit] =
     for {
       ctx <- ev.ask
 
-      initialGoogleCall = msg.cluster
-        .traverse(createClusterMessage =>
-          gkeInterp.createCluster(createClusterMessage.clusterId, List(createClusterMessage.defaultNodepoolId), false)
-        )
-      clusterResult <- initialGoogleCall.adaptError {
-        case e =>
-          PubsubKubernetesError(
-            AppError(e.getMessage(), ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Cluster, None),
-            Some(msg.appId),
-            false,
-            msg.cluster.map(_.defaultNodepoolId),
-            msg.cluster.map(_.clusterId)
-          )
-      }
-
-      createClusterAndNodepool = for {
-        _ <- clusterResult.traverse(result => gkeInterp.pollCluster(result)).adaptError {
+      // The "create app" flow does a number of things:
+      //  1. (optional) create, poll, and setup cluster
+      //  2. (optional) create and poll nodepool
+      //  3. (optional) create and poll disk
+      //  4. create and poll app
+      //
+      // Numbers 1-3 are all Google calls; (4) is a helm call. If either of (1) or (2) are
+      // necessary then we will do the _initial_ Google call synchronous to the pubsub processing so
+      // we can nack the message on errors. Monitoring all creations will be asynchronous,
+      // and (3) and (4) will always be asynchronous.
```

Comment:
Trying to understand why we're not doing the initial Google call synchronous to the pubsub processing, for (3) also, so we can nack the message on errors for that as well.

Reply:
Yeah, the original code (`createDiskForApp`) had disk creation + monitoring async, so I decided not to change it. We COULD refactor to make the initial call sync, but I feel like it's not as important -- the GKE calls are the ones we want to nack and retry on errors.